src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,53 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final Color DEFAULT_COLOR = Color.WHITE;
+    private static final int DEFAULT_RADIUS = 10;
+    private static final int MAX_RANGE = 20;
+    private final Random random = new Random();
+    private final Circle circle = new Circle();
+    private final Square square = new Square();
+    private final RightTriangle rightTriangle = new RightTriangle();
+    private final Rectangle rectangle = new Rectangle();
+    private final IsoscelesTrapezoid isoscelesTrapezoid = new IsoscelesTrapezoid();
```

Comment:
Don't create them here. See the example below.

You will always return a reference to one single object. Instead, you need a new object every time you call "getRandomFigure".